php_phongo.c

```diff
@@ -1435,6 +1435,31 @@ static bool php_phongo_uri_finalize_auth(mongoc_uri_t* uri) /* {{{ */
 	return true;
 } /* }}} */
 
+static bool php_phongo_uri_finalize_directconnection(mongoc_uri_t* uri) /* {{{ */
+{
+	const mongoc_host_list_t* hosts;
+
+	if (!mongoc_uri_get_option_as_bool(uri, MONGOC_URI_DIRECTCONNECTION, false)) {
+		return true;
+	}
+
+	/* Per the URI options spec, directConnection conflicts with multiple hosts
+	 * and SRV URIs, which may resolve to multiple hosts. */
+	if (!strncmp(mongoc_uri_get_string(uri), "mongodb+srv://", 14)) {
```

Comment:
This code was lifted from `mongoc_uri_parse_scheme`, as libmongoc doesn't have a public API to report if SRV is used.